tests/training/metrics_test.py

```diff
@@ -0,0 +1,139 @@
+                             [1, 0, 1]])
+        accuracy(predictions, targets, mask)
+        actual_accuracy = accuracy.get_metric(reset=True)
+        numpy.testing.assert_almost_equal(actual_accuracy["accuracy"], 50.0)
+
+    def test_top_k_categorical_accuracy_catches_exceptions(self):
+        accuracy = CategoricalAccuracy()
+        predictions = torch.rand([5, 7])
+        out_of_range_labels = torch.Tensor([10, 3, 4, 0, 1])
+        with pytest.raises(ConfigurationError):
+            accuracy(predictions, out_of_range_labels)
+
+    def test_f1_measure(self):
+        f1_measure = F1Measure(null_prediction_label=0)
+        predictions = torch.Tensor([[0.35, 0.25, 0.1, 0.1, 0.2],
+                                    [0.1, 0.6, 0.1, 0.2, 0.0],
+                                    [0.1, 0.6, 0.1, 0.2, 0.0],
+                                    [0.1, 0.5, 0.1, 0.2, 0.0],
+                                    [0.1, 0.2, 0.1, 0.7, 0.0],
+                                    [0.1, 0.6, 0.1, 0.2, 0.0]])
+        # [True Negative, False Negative, True Positive,
+        #  True Positive, True Positive, False Negative]
+        targets = torch.Tensor([0, 5, 1, 0, 3, 0])
+        f1_measure(predictions, targets)
+        f1_metrics = f1_measure.get_metric()
+        assert f1_measure.true_positives == 2.0
+        assert f1_measure.true_negatives == 1.
+        assert f1_measure.false_positives == 1.0
+        assert f1_measure.false_negatives == 2.0
+        f1_measure.reset()
+        numpy.testing.assert_almost_equal(f1_metrics["precision"], 0.666666666)
+        numpy.testing.assert_almost_equal(f1_metrics["recall"], 0.5)
+        numpy.testing.assert_almost_equal(f1_metrics["f1-measure"], 0.57142857)
+
+        # Test the same thing with a mask:
+        mask = torch.Tensor([0, 1, 1, 1, 1, 0])
+        f1_measure(predictions, targets, mask)
+        f1_metrics = f1_measure.get_metric()
+        assert f1_measure.true_positives == 2.0
+        assert f1_measure.true_negatives == 0.0
+        assert f1_measure.false_positives == 1.0
+        assert f1_measure.false_negatives == 1.0
+        f1_measure.reset()
+        numpy.testing.assert_almost_equal(f1_metrics["precision"], 0.666666666)
+        numpy.testing.assert_almost_equal(f1_metrics["recall"], 0.6666666666)
+        numpy.testing.assert_almost_equal(f1_metrics["f1-measure"], 0.6666666666)
+
+
+    def test_f1_measure_works_for_sequences(self):
+        f1_measure = F1Measure(null_prediction_label=0)
+        predictions = torch.Tensor([[[0.35, 0.25, 0.1, 0.1, 0.2],
+                                     [0.1, 0.6, 0.1, 0.2, 0.0],
+                                     [0.1, 0.6, 0.1, 0.2, 0.0]],
+                                    [[0.35, 0.25, 0.1, 0.1, 0.2],
+                                     [0.1, 0.6, 0.1, 0.2, 0.0],
+                                     [0.1, 0.6, 0.1, 0.2, 0.0]]])
+        # [[True Negative, False Positive, False Positive],
```

Comment:
I think multi-class F1 is just a confusing metric that we probably don't need to worry about.  What you really want to do is compute F1 for each (non-null) label independently, and report that, or some aggregation of the individual F1 measures.  We should actually implement a span-based F1 for SRL.  But, all of that can be in a separate PR.  I'd recommend slightly modifying your F1 measure to take the positive label (instead of the null label), and compute F1 of that label.  Then it can be pretty easily be used directly for binary classification, and if a model wants to instantiate multiples of these for each label, and compute some aggregation, they can.